### DISCUSSION TITLE
Multiple relationships between containers of 2 systems do not render properly - only the last one declared is rendered

### DISCUSSION THREAD

**Discussion by yvesaston** [structurizr/lite#54] [Q&A] [answered]
### Description

When there are 2 relationships between containers of 2 software systems (for example because the technology underpinning those relationships is not the same), only the last relationship declared in the DSL is rendered ; expectation is to have both relationships rendered.
Example 1 (relationship a.c2 -> b.c2 not rendered) :
```
 workspace {
    !identifiers hierarchical
    model {
          a = softwareSystem "System A" {
              c1 = container "Container A1"
              c2 = container "Container A2"
          }
          b = softwareSystem "System B" {
              c1 = container "Container B1"
              c2 = container "Container B2"
          }
          a.c1 -> b.c1 "Does 1" "Techno 1"
          a.c2 -> b.c2 "Does 2" "Techno 2"
    }
    views {
        systemLandscape test-ab "Test 4 bug"
            include *
            autolayout lr
    }
}
```

On the contrary, if:
-  one relationship is between containers of those 2 systems,
-  and one relationship is between the 2 systems directly
Then the 2 relationships are rendered as expected.
Example 2 (relationship a -> b rendered):
```
workspace {
    !identifiers hierarchical
    model {
          a = softwareSystem "System A" {
              c1 = container "Container A1"
              c2 = container "Container A2"
          }
          b = softwareSystem "System B" {
              c1 = container "Container B1"
              c2 = container "Container B2"
          }
          a.c1 -> b.c1 "Does 1" "Techno 1"
          a -> b "Does 2" "Techno 2"
    }
    views {
        systemLandscape test-ab "Test 4 bug"
            include *
            autolayout lr
    }
}
```

### Steps to reproduce

Render the Description' Example 1 in Structurizr/lite

### Configuration

Docker for Structurizr/lite 3046.

### Severity

Major

### Priority

Medium

### Resolution

I have no budget, please fix this for free

**Comment by simonbrowndotje** [comment, marked as answer]
It sounds like you're seeing the implied relationships feature in action -> https://github.com/structurizr/dsl/tree/master/docs/cookbook/implied-relationships

**Reply by yvesaston** [reply]
@simonbrowndotje , thanks for your very fast reaction ! Got it ; red the cookbook but I still think there is value in presenting those 2 relationships separately, mainly because the technologies used are different (and we are doing system architecture, not business 😊).
So I have clear that in that case I need to declare in my workspace both the "real" relationship and the "implied" one/s. That's good enough.

**Reply by simonbrowndotje** [reply]
You can also plug-in a different strategy for creating implied relationships (and create your own if required); for example:

```
 workspace {
    !identifiers hierarchical

    !script groovy {
        workspace.model.impliedRelationshipsStrategy = new com.structurizr.model.CreateImpliedRelationshipsUnlessSameRelationshipExistsStrategy()
    }
    
    model {
          a = softwareSystem "System A" {
              c1 = container "Container A1"
              c2 = container "Container A2"
          }
          b = softwareSystem "System B" {
              c1 = container "Container B1"
              c2 = container "Container B2"
          }
          a.c1 -> b.c1 "Does 1" "Techno 1"
          a.c2 -> b.c2 "Does 2" "Techno 2"
    }
    views {
        systemLandscape test-ab "Test 4 bug"
            include *
            autolayout lr
    }
}
```

This results in:

![image](https://user-images.githubusercontent.com/1009874/229821279-2adf41ba-8aca-4416-b670-9b6ac358df27.png)
